proto/api_v2/metricsquery.proto

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax="proto3";
+
+package jaeger.api_v2;
+
+import "model.proto";
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "opentelemetry/proto/metrics/v1/metrics.proto";
+
+option go_package = "api_v2";
+option java_package = "io.jaegertracing.api_v2";
+
+// MetricsQueryBaseRequest is the base request parameter accompanying a MetricsQueryService RPC call.
+message MetricsQueryBaseRequest {
+  // end_time is the ending time of the time series query range.
+  // Optional. Default = now.
+  google.protobuf.Timestamp end_time = 1 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = true
+  ];
+
+  // lookback is the duration from the end_time to look back on for metrics data points.
+  // For example, if set to 1h, the query would span from end_time-1h to end_time.
+  // Optional. Default = 1h.
+  google.protobuf.Duration lookback = 2 [
+    (gogoproto.stdduration) = true,
+    (gogoproto.nullable) = true
+  ];
+
+  // step size is the duration between data points of the query results.
+  // For example, if set to 5s, the results would produce a data point every 5 seconds
+  // from the start_time to end_time.
+  // Optional. Default = 5s.
+  google.protobuf.Duration step = 3 [
```

Comment:
Why do you need this? Why not just accept whichever precision the backend can return?

Reply:
Required by [M3's query API](https://m3db.io/docs/reference/m3query/api/query/):
```
curl -X "POST" -G "http://localhost:7201/api/v1/query_range" \
   -d "query=calls_total" \
   -d "start=$( date -v -45S +%s )" \
   -d "end=$( date +%s )" | jq .
   
{
  "status": "error",
  "error": "error parsing param: step, error: expected positive step size, instead got: 0"
}
```

Further, I think it's useful for callers to control the amount of data downloaded from the query engine. 

What if metrics data is written at a higher precision in the timeseries db backend, or the timeseries backend is able to interpolate data points at a higher default precision, than the caller desires because it doesn't need such a high precision at the cost of performance?

Reply:
Ok. But I'm concerned with the reverse, client asking for precision that server doesn't have, what is the expected behavior? This param can only be satisfied if it's equal to or a multiple of the server's native precision, and how is client supposed to know that?

Reply:
The client of the API doesn't know about the backend implementation. The only possible approach I see is if there is another api call that returns the supported precisions and the ui makes the user choose